### ISSUE TITLE
Session cookie is lost after redirecting

### ISSUE THREAD

**Issue by jonasraoni** [cypress-io/cypress#25495]
### Current behavior

Considering a simple:
```js
cy.visit('/');
```

And the HTTP response below:

```http
HTTP/1.1 302 Found
Host: 127.0.0.1:8080
Date: Sun, 08 Jan 2023 16:07:27 GMT
Connection: close
Expires: Thu, 19 Nov 1981 08:52:00 GMT
Cache-Control: no-store, no-cache, must-revalidate
Pragma: no-cache
Set-Cookie: SESSION_ID=deleted; expires=Thu, 01-Jan-1970 00:00:01 GMT; Max-Age=0
Set-Cookie: SESSION_ID=new_cookie_value
Location: next-url
Content-type: text/html; charset=UTF-8
```

> Notice the `Set-Cookie`, first removing a cookie, then recreating it, and the presence of the `Location` header

When Cypress goes to the redirected URL, the cookie is lost.

If I remove the `Location` header from the response and redirect manually with another `cy.visit('/next-url');` OR if I drop the first `Set-Cookie` header (the one which deletes the cookie), then everything works fine.

### Desired behavior

I expect Cypress to behave like the browser, which understands the last `Set-Cookie` should keep/recreate the cookie.

### Test code to reproduce

Small sample available above, I might fork the https://github.com/cypress-io/cypress-test-tiny later, but definitely not at this moment.

### Cypress Version

12.2.0

### Node version

18.11.0

### Operating System

Windows 10

### Other

Original report is here: https://github.com/cypress-io/cypress/issues/19800#issuecomment-1374876892

**Comment by AtofStryker** [comment]
Hey @jonasraoni. Hope you have been well and thank you for opening an issue. Would you be able to help me get a minimum reproduction repository up and running? I don't think it has to be much. Maybe visiting a webpage that sets the session cookies you expect and then run it in the browser vs Cypress? I think you might be able to do that with cypress-test-tiny.

I do wonder if it works without a `302` redirect but the reproduction will help us confirm.

**Comment by jonasraoni** [comment]
> Would you be able to help me get a minimum reproduction repository up and running?

I'll take a look later, for now I'm a bit overloaded to inspect other repos. I just can confirm it happened with another person using another SO/environment.

> I do wonder if it works without a 302 redirect but the reproduction will help us confirm

Yes, I wrote about it above, if I drop the `Location` header, it works fine. Redirecting manually with another `cy.visit()` also works.

**Comment by DobQA** [comment]
Im seeing something similar, are you only getting this issue on Chromium based browsers. For myself no issues firefox and Safari, but rest have problem?

**Comment by suellenzsoares** [comment]
Hello @AtofStryker,
I believe I'm having a similar issue, after be already logged and the tests is executing it just logout.
I really appreciate if you can help me with it once it is blocking our regression tests.

Thank you so much

**Comment by mschile** [comment]
@jonasraoni, I tried to recreate the issue using this [repo](https://github.com/mschile/cypress-test-tiny/tree/issue-25495) but I was unable to reproduce it. Please let me know if I am missing something.

@suellenzsoares, do you have a project that I could look at or can you update this [repo](https://github.com/mschile/cypress-test-tiny/tree/issue-25495) to reproduce the issue.

**Comment by suellenzsoares** [comment]
Hello!

Can it be remotely? Because I don't have permission to share the code.

Thank you for fast reply!

**Comment by mschile** [comment]
Possibly related to #25205 

@jonasraoni, here is potential [workaround](https://github.com/cypress-io/cypress/issues/25205#issuecomment-1416873663).

@suellenzsoares, unfortunately, I won't be able to help remotely but please feel free to try the above workaround if it's the same issue. Otherwise, would you be able to recreate your issue with a reproduction repository by forking [cypress-test-tiny](https://github.com/cypress-io/cypress-test-tiny) or by other means?

**Comment by denissonp** [comment]
Also, it seems like, when the same cookie is recreated after a redirecting, Cypress is keeping both cookies, instead of only keeping the last one created.
At least, this is the behaviour I get. and If I manually delete the old duplicated cookies from the browser, it works just fine.
I'm using Chrome to test, btw.

**Comment by suellenzsoares** [comment]
I didn't have time yet to try it this week, I'm executing other tests, as
soon as I have time I'll try again and let you all know.

**Comment by AtofStryker** [comment]
@jonasraoni @denissonp @suellenzsoares @DobQA would any of you be able to try [this binary](https://github.com/cypress-io/cypress/commit/86a8cd83792a046146bb43d869f8812dd0fc800c) to see if it resolves your issue?

**Comment by AtofStryker** [comment]
This should be fixed with 12.6.1, which should be released either today or tomorrow.
